universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -47,12 +50,77 @@ public static Builder builder() {
   /** A human readable description of a Program, localized for each supported locale. */
   public abstract LocalizedStrings localizedDescription();
 
-  /** The list of {@link BlockDefinition}s that make up the program. */
   public abstract ImmutableList<BlockDefinition> blockDefinitions();
 
   /** The list of {@link ExportDefinition}s that make up the program. */
   public abstract ImmutableList<ExportDefinition> exportDefinitions();
 
+  /**
+   * Returns a program definition with block definitions such that each enumerator block is
+   * immediately followed by all of its repeated and nested repeated blocks. This method should be
+   * used when {@link #hasOrderedBlockDefinitions()} is a precondition for manipulating blocks.
+   *
+   * <p>Programs created before early June 2021 may not satisfy this condition.
+   */
+  public ProgramDefinition orderBlockDefinitions() {
+    if (!hasOrderedBlockDefinitions()) {
+      ProgramDefinition orderedProgramDefinition =
+          toBuilder()
+              .setBlockDefinitions(reorderBlockDefinitionsInner(getNonRepeatedBlockDefinitions()))
+              .build();
+      orderedProgramDefinition.hasOrderedBlockDefinitionsMemo = true;
+      return orderedProgramDefinition;
+    }
+    return this;
+  }
+
+  private ImmutableList<BlockDefinition> reorderBlockDefinitionsInner(
```

Comment:
nit: rename to match the new name of the public method? (reoder --> order). And maybe same with models.Program.java

Reply:
done!